utils/doclinter.py

```diff
@@ -1,77 +0,0 @@
-"""A linter for Sphinx docs"""
-
-import os
-import re
-import sys
-from typing import List
-
-MAX_LINE_LENGTH = 85
```

Comment:
It seems `sphinx-lint` does not check the length of lines. So I'd not like to remove this script from our repository. How about using both?

Reply:
Perhaps `sphinx-lint` could add a line-length option?

Otherwise we could simplify to e.g.:

```python
"""Check line length in rst documents."""

import os
import re
import sys

MAX_LINE_LENGTH = 85


def check_line_length(path: str) -> bool:
    with open(path) as f:
        document = f.readlines()

    error = False
    for i, line in enumerate(document):
        if len(line) > MAX_LINE_LENGTH:
            if (
                re.match(r'^\s*\.\. ', line)  # ignore directives and hyperlink targets
                or re.match(r'^\s*__ ', line)  # ignore anonymous hyperlink targets
                or re.match(r'^\s*``[^`]+``$', line)  # ignore a very long literal string
            ):
                pass
            else:
                print(f'{path}:{i + 1}: the line is too long ({len(line)} > {MAX_LINE_LENGTH}).')
                error = True

    return error


if __name__ == '__main__':
    error = False
    for path in sys.argv[1:]:
        if os.path.isfile(path):
            error ^= check_line_length(path)
        elif os.path.isdir(path):
            for root, _dirs, files in os.walk(path):
                for filename in files:
                    if filename.endswith('.rst'):
                        path = os.path.join(root, filename)
                        error ^= check_line_length(path)
    raise SystemExit(error + 0)
```

A

Reply:
I'd not like to reduce our checks for docs. So it's very nice if sphinx-lint support it. Until then, it's also okay to remove duplicated features from doclinter.py.

Reply:
Ohhh yes good catch.

We do have this checker, but it's not enabled by default:

https://github.com/sphinx-contrib/sphinx-lint/blob/b25d09c5a0cf15030c575d18078d815724a1e3e0/sphinxlint.py#L316

And if we enable it using `--severity 0` it'll whine about other things that you typically allow like using default roles.

We'll have to implement a way to enable/disable checks by name instead of by severity level for more granularity.

I'll be offline the next week, but there's no rush anyway.